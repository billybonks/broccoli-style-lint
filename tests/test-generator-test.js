/* eslint-disable no-useless-escape*/
let outputs ={
  qunit: {
    errored:
`QUnit.module('Stylelint');
QUnit.test('has-errors.scss should pass stylelint', function(assert) {
  assert.expect(1);
  assert.ok(false, '1:15 Unexpected empty block (block-no-empty)\\n6:10 Expected \\\\\"#000000\\\\\" to be \\\\\"black\\\\\" (color-named)');
});
`,
  passed:
`QUnit.module('Stylelint');
QUnit.test('no-errors.scss should pass stylelint', function(assert) {
  assert.expect(1);
  assert.ok(true, 'no-errors.scss should pass stylelint');
});
`
},
  mocha: {
    errored:
`describe('Stylelint', function() {
  it('has-errors.scss should pass stylelint', function() {
    // test failed
    var error = new chai.AssertionError('1:15 Unexpected empty block (block-no-empty)\\n6:10 Expected \\\\\"#000000\\\\\" to be \\\\\"black\\\\\" (color-named)');
    error.stack = undefined;
    throw error;
  });
});
`,
    passed:
`describe('Stylelint', function() {
  it('no-errors.scss should pass stylelint', function() {
    // test passed
  });
});
`
  }
};

const frameworks = ['qunit', 'mocha'];
const generator = require('../lib/test-generator');
const errors = require('./fixtures/errors');

describe('test-generator', function() {
  describe('errors is defined', function() {
    frameworks.forEach(function(framework){
      describe(framework, function() {
        it('generates correct errored test', function(){
          let results = generator('has-errors.scss', errors, framework);
          expect(results).toEqual(outputs[framework].errored);
        });
        it('generates correct passed test', function(){
          let results = generator('no-errors.scss', null, framework);
          expect(results).toEqual(outputs[framework].passed);
        });
      });
    });
  });
});
