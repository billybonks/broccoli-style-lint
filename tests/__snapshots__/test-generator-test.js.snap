// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`suite-generator errors is defined mocha generates correct errored test 1`] = `
"  it('has-errors.scss should pass stylelint', function() {
    // test failed
    var error = new chai.AssertionError('1:15 Unexpected empty block (block-no-empty)\\\\n6:10 Expected \\\\\\\\\\"#000000\\\\\\\\\\" to be \\\\\\\\\\"black\\\\\\\\\\" (color-named)');
    error.stack = undefined;
    throw error;
  });
"
`;

exports[`suite-generator errors is defined mocha generates correct passed test 1`] = `
"  it('no-errors.scss should pass stylelint', function() {
    // test passed
  });
"
`;

exports[`suite-generator errors is defined qunit generates correct errored test 1`] = `
"QUnit.test('has-errors.scss should pass stylelint', function(assert) {
  assert.expect(1);
  assert.ok(false, '1:15 Unexpected empty block (block-no-empty)\\\\n6:10 Expected \\\\\\\\\\"#000000\\\\\\\\\\" to be \\\\\\\\\\"black\\\\\\\\\\" (color-named)');
});
"
`;

exports[`suite-generator errors is defined qunit generates correct passed test 1`] = `
"QUnit.test('no-errors.scss should pass stylelint', function(assert) {
  assert.expect(1);
  assert.ok(true, 'no-errors.scss should pass stylelint');
});
"
`;

exports[`test-generator errors is defined mocha generates correct errored test 1`] = `
"  it('has-errors.scss should pass stylelint', function() {
    // test failed
    var error = new chai.AssertionError('1:15 Unexpected empty block (block-no-empty)\\\\n6:10 Expected \\\\\\\\\\"#000000\\\\\\\\\\" to be \\\\\\\\\\"black\\\\\\\\\\" (color-named)');
    error.stack = undefined;
    throw error;
  });
"
`;

exports[`test-generator errors is defined mocha generates correct passed test 1`] = `
"  it('no-errors.scss should pass stylelint', function() {
    // test passed
  });
"
`;

exports[`test-generator errors is defined qunit generates correct errored test 1`] = `
"QUnit.test('has-errors.scss should pass stylelint', function(assert) {
  assert.expect(1);
  assert.ok(false, '1:15 Unexpected empty block (block-no-empty)\\\\n6:10 Expected \\\\\\\\\\"#000000\\\\\\\\\\" to be \\\\\\\\\\"black\\\\\\\\\\" (color-named)');
});
"
`;

exports[`test-generator errors is defined qunit generates correct passed test 1`] = `
"QUnit.test('no-errors.scss should pass stylelint', function(assert) {
  assert.expect(1);
  assert.ok(true, 'no-errors.scss should pass stylelint');
});
"
`;
